### PR TITLE
Syntactic check testing and minor refactoring of cfg structure

### PR DESCRIPTION
**Pull request by lucaneg**
**Content**
* A more structured test for `SyntacticCheck` application has been implemented. Small bugs in the exection engine have been fixed.
* `CFG` class now has a `isEqualTo(CFG)` method for content-equality among different instances.
* `Statement` instances now rely on reference equality, with an `isEqualTo(Statement)` for content-equality among instances.
* Small tests for logging utilities.

### CODE REVIEW

File: lisa/.gitignore [hunk 1]
```diff
@@ -42,3 +42,4 @@ local.properties
 
 # UML diagrams
 diagrams/
+/lisa/
```
Comment (VincenzoArceri) [inline lisa/.gitignore:45]: We should remove this (maybe added again by my previous commits)

**Review by VincenzoArceri — CHANGES_REQUESTED** [review]
Minor comments to be checked

File: lisa/.gitignore (hunk 1 above)
Comment (lucaneg) [reply lisa/.gitignore:45]: Yes, it got added with a previous commit. Removed.

File: lisa/src/main/java/it/unive/lisa/cfg/statement/NativeCall.java [hunk 2]
```diff
@@ -53,12 +55,35 @@ public NativeCall(CFG cfg, String sourceFile, int line, int col, String construc
...
+	public boolean isEqualTo(Statement st) {
+		if (this == st)
+			return true;
+		if (getClass() != st.getClass())
+			return false;
+		NativeCall other = (NativeCall) st;
+		if (constructName == null) {
+			if (other.constructName != null)
+				return false;
+		} else if (!constructName.equals(other.constructName))
```
[the remark this answers is not in the document]
Comment (lucaneg) [reply lisa/src/main/java/it/unive/lisa/cfg/statement/NativeCall.java:80]: There was a missing call to super.isEqualTo(other) at the end. Good catch!

File: lisa/src/main/java/it/unive/lisa/cfg/statement/OpenCall.java [hunk 3]
```diff
@@ -56,6 +58,29 @@ public String getTargetName() {
...
+	public boolean isEqualTo(Statement st) {
+		if (this == st)
+			return true;
+		if (getClass() != st.getClass())
+			return false;
+		OpenCall other = (OpenCall) st;
+		if (targetName == null) {
+			if (other.targetName != null)
+				return false;
+		} else if (!targetName.equals(other.targetName))
```
[the remark this answers is not in the document]
Comment (lucaneg) [reply lisa/src/main/java/it/unive/lisa/cfg/statement/OpenCall.java:79]: Same as the other comment

**Review by VincenzoArceri — APPROVED** [review]
Perfect!

**Pull request merged**